lib/qos/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+target_include_directories(app PRIVATE .)
+target_sources(app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/qos.c)
```

Comment:
This doesn't seem to be right, you should rather create a separate target for the library, please have a look at https://github.com/nrfconnect/sdk-nrf/blob/main/lib/at_shell/CMakeLists.txt for example.